build-maven/build.sh

```diff
@@ -0,0 +1,336 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Required environment variables
+: "${ARTIFACTORY_URL:="https://repox.jfrog.io/artifactory"}"
+: "${ARTIFACTORY_DEPLOY_REPO:?}" "${ARTIFACTORY_DEPLOY_PASSWORD:?}" "${ARTIFACTORY_PRIVATE_PASSWORD:?}"
+: "${GITHUB_REF_NAME:?}" "${BUILD_NUMBER:?}" "${GITHUB_REPOSITORY:?}"
+: "${GITHUB_EVENT_NAME:?}" "${PULL_REQUEST:?}" "${PULL_REQUEST_SHA:-}"
+: "${SONAR_HOST_URL:?}" "${SONAR_TOKEN:?}"
+
+# Set BUILD_ID for compatibility with SonarSource parent POM to change when BUILD-8508 resolved
+BUILD_ID=$BUILD_NUMBER
+export BUILD_NUMBER BUILD_ID
+
+# =============================================================================
+# CONFIGURATION CONSTANTS
+# =============================================================================
+
+readonly SCANNER_VERSION="${SCANNER_VERSION:-5.0.0.4389}"
+readonly MASTER_MAVEN_OPTS="-Xmx1536m -Xms128m"
+readonly MAINTENANCE_MAVEN_OPTS="-Xmx1536m -Xms128m"
+readonly PR_MAVEN_OPTS="-Xmx1G -Xms128m"
```

Comment:
this sounds like a debt: was there a reason for different parameters?
I would keep only one common MAVEN_OPTS